HoloJS/HoloJsHost/RenderingContext2D.cpp

```diff
@@ -132,18 +357,25 @@ void RenderingContext2D::getImageDataBGRFlipY(vector<byte>& bgrPixels)
     auto canvasPixelsLength = canvasPixels->Length;
     auto canvasPixelData = canvasPixels->begin();
 
-    const int dest_bpp = 3;
+    const int dest_bpp = 4;
     bgrPixels.resize((canvasPixelsLength * dest_bpp) / m_bpp);
 
     // Convert from 32bpp RGBA to 24bpp BGR and flip vertical
     for (int row_index = 0; row_index < m_height; row_index++) {
         for (int column_index = 0; column_index < m_width; column_index++) {
             const auto destination_row_offset = row_index * m_width * dest_bpp;
-            const auto source_row_offset = (m_height - row_index - 1) * m_width * m_bpp;
-
-            bgrPixels[destination_row_offset + column_index * dest_bpp + 0] = canvasPixelData[source_row_offset + column_index * m_bpp + 2];
-            bgrPixels[destination_row_offset + column_index * dest_bpp + 1] = canvasPixelData[source_row_offset + column_index * m_bpp + 1];
-            bgrPixels[destination_row_offset + column_index * dest_bpp + 2] = canvasPixelData[source_row_offset + column_index * m_bpp + 0];
+            const auto source_row_offset = (row_index)*m_width * m_bpp;
+
+            unsigned char alpha = canvasPixelData[source_row_offset + column_index * m_bpp + 3];
+            // un-premultiply the alpha values, because the image encoders wants un-premultiplied alpha.
```

Comment:
This should not be un-premultiplied since the JPEG encoder only accepts 24bpp and we want the alpha to be used for the encoded image.

This method is used by the toDataURL function for getting a base64 encoded PNG or JPEGimage that can be uploaded or shared outside the process.